instrumentation/opentelemetry-instrumentation-aiohttp-server/src/opentelemetry/instrumentation/aiohttp_server/__init__.py

```diff
@@ -296,7 +294,22 @@ class AioHttpServerInstrumentor(BaseInstrumentor):
     """
 
     def _instrument(self, **kwargs):
+        tracer_provider = kwargs.get("tracer_provider", None)
+        assert tracer_provider is None or isinstance(
+            tracer_provider, trace.TracerProvider
+        )
         self._original_app = web.Application
+
+        _middleware = create_aiohttp_middleware(tracer_provider=tracer_provider)
+        class _InstrumentedApplication(web.Application):
```

Comment:
Could we either keep `class _InstrumentedApplication` at top level, or implement a factory? Else adds some complexity.